toolkit/tools/pkg/imagecustomizerlib/createuuid.go

```diff
@@ -47,3 +51,61 @@ func convertUuidToString(uuid [UuidSize]byte) string {
 
 	return uuidStr
 }
+
+func extractImageUUID(imageConnection *ImageConnection) ([UuidSize]byte, string, error) {
+	var emptyUuid [UuidSize]byte
+
+	releasePath := filepath.Join(imageConnection.Chroot().RootDir(), "etc/image-customizer-release")
+	data, err := file.Read(releasePath)
+	if err != nil {
+		return emptyUuid, "", fmt.Errorf("failed to read %s:\n%w", releasePath, err)
+	}
+
+	lines := strings.Split(string(data), "\n")
+	var uuidStr string
+	for _, line := range lines {
+		if strings.HasPrefix(line, "IMAGE_UUID=") {
+			uuidStr = strings.Trim(strings.TrimPrefix(line, "IMAGE_UUID="), `"`)
+			break
+		}
+	}
+
+	if uuidStr == "" {
+		return emptyUuid, "", fmt.Errorf("IMAGE_UUID not found in %s", releasePath)
+	}
+
+	parsed, err := parseUuidString(uuidStr)
+	if err != nil {
+		return emptyUuid, "", fmt.Errorf("failed to parse IMAGE_UUID from string %s:\n%w", uuidStr, err)
```

Comment:
nit: `"failed to parse IMAGE_UUID (%s):\n%w"`

Reply:
Fixed.